homeassistant/components/broadlink/__init__.py

```diff
@@ -3,29 +3,49 @@
 
 from .const import DOMAIN
 from .device import BroadlinkDevice
+from .discovery import BroadlinkDiscovery
 
 
 @dataclass
 class BroadlinkData:
-    """Class for sharing data within the Broadlink integration."""
+    """Class for sharing data in the Broadlink integration."""
 
+    config: dict = None
+    discovery: BroadlinkDiscovery = None
     devices: dict = field(default_factory=dict)
     platforms: dict = field(default_factory=dict)
 
 
 async def async_setup(hass, config):
     """Set up the Broadlink integration."""
-    hass.data[DOMAIN] = BroadlinkData()
+    config = config.get(DOMAIN)
+    discovery = BroadlinkDiscovery(hass)
+    hass.data[DOMAIN] = BroadlinkData(config, discovery)
+    await discovery.async_setup()
```

Comment:
We do this in config entry setup so we don't need this here.

Reply:
The idea here is to allow people to discover devices even before one is configured. Without this, they would need to know the first IP address to add the device via Integrations and only then have access to device discovery. They can permanently activate device discovery by adding this line to configuration.yaml:
```yaml
broadlink:
```
The alternative would be adding a step in the configuration flow to discover new devices there.

Reply:
We shouldn't rely on anything in the YAML config.

Reply:
Ok, I will make this better.